rapidproto/rapidproto.go

```diff
@@ -15,20 +17,70 @@ func MessageGenerator[T proto.Message](x T, options GeneratorOptions) *rapid.Gen
 	return rapid.Custom(func(t *rapid.T) T {
 		msg := msgType.New()
 
-		options.setFields(t, msg, 0)
+		options.setFields(t, nil, msg, 0)
 
 		return msg.Interface().(T)
 	})
 }
 
 type GeneratorOptions struct {
-	AnyTypeURLs []string
-	Resolver    protoregistry.MessageTypeResolver
+	AnyTypeURLs    []string
+	InterfaceHints map[string]string
+	Resolver       protoregistry.MessageTypeResolver
+
+	// NoEmptyLists will cause the generator to not generate empty lists
+	// Recall that an empty list will marshal (and unmarshal) to null.   Some encodings may treat these states
+	// differently.  For example, in JSON, an empty list is encoded as [], while null is encoded as null.
+	NoEmptyLists bool
+
+	// DisallowNilMessages will cause the generator to not generate nil messages to protoreflect.MessageKind fields
+	DisallowNilMessages bool
+
+	// GogoUnmarshalCompatibleDecimal will cause the generator to generate decimal values which, when serialized as
+	// either bytes or a string, are compatible with gogo's unmarshaler.  These fields are identified by the presence
+	// of (amino.encoding = "cosmos_dec_bytes") or (cosmos_proto.scalar = "cosmos.Dec") annotations.
+	// If this option is set, the AminoEncodingExtension must also be set.
+	GogoUnmarshalCompatibleDecimal bool
```

Comment:
Can't we just have some way of registering special handlers for amino.encoding and cosmos_proto.scalar with the generator?

Reply:
That would work too, although I'm not sure it's any simpler.

Reply:
On the other hand, it would remove this horribly specific field name.

Reply:
I also prefer this new FieldMapper, it removes the dep on SDK concepts such as aminoExtension and cosmos.Dec